iocore/net/SSLCertLookup.cc

```diff
@@ -471,11 +492,12 @@ SSLContextStorage::lookup(const char *name) const
   }
 
   // Then strip off the top domain name and look for a wildcard domain match
-  const char *subdomain = index(name, '.');
+  const char *subdomain = index(lower_case_name, '.');
```

Comment:
Was this a bug in the previous code? It seems this changes semantics as well as changing to the HashMap, no?